app/src/sharedTest/java/org/oppia/app/player/audio/AudioFragmentTest.kt

```diff
@@ -0,0 +1,255 @@
+  }
+
+  @Test
+  fun testAudioFragment_invokePrepared_clickPlay_touchSeekBar_checkStillPlaying() {
+    shadowMediaPlayer.invokePreparedListener()
+
+    onView(withId(R.id.ivPlayPauseAudio)).perform(click())
+    onView(withId(R.id.sbAudioProgress)).perform(clickSeekBar(100))
+
+    onView(withId(R.id.ivPlayPauseAudio)).check(matches(withContentDescription(R.string.audio_pause_description)))
+  }
+
+
+  @Test
+  fun testAudioFragment_invokePrepared_playAudio_configurationChange_checkStillPlaying() {
+    shadowMediaPlayer.invokePreparedListener()
+    onView(withId(R.id.ivPlayPauseAudio)).perform(click())
+    onView(withId(R.id.sbAudioProgress)).perform(clickSeekBar(100))
+
+    activityScenario.onActivity { activity ->
+      activity.requestedOrientation = Configuration.ORIENTATION_LANDSCAPE
+    }
+
+    onView(withId(R.id.ivPlayPauseAudio)).check(matches(withContentDescription(R.string.audio_pause_description)))
+  }
+
+  @Test
+  fun testAudioFragment_invokePrepared_changeDifferentLanguage_checkResetSeekBarAndPaused() {
+    shadowMediaPlayer.invokePreparedListener()
+    onView(withId(R.id.ivPlayPauseAudio)).perform(click())
+    onView(withId(R.id.sbAudioProgress)).perform(clickSeekBar(100))
+
+    onView(withId(R.id.tvAudioLanguage)).perform(click())
+    onView(withText("es")).inRoot(isDialog()).perform(click())
+    onView(withText("OK")).inRoot(isDialog()).perform(click())
+
+    onView(withId(R.id.ivPlayPauseAudio)).check(matches(withContentDescription(R.string.audio_play_description)))
+    onView(withId(R.id.sbAudioProgress)).check(matches(withSeekBarPosition(0)))
+  }
+
+  private fun withContentDescription(contentDescriptionId: Int) = object : TypeSafeMatcher<View>() {
+    private val contentDescription = context.getString(contentDescriptionId)
+
+    override fun describeTo(description: Description) {
+      description.appendText("ImageView with contentDescription same as $contentDescription")
+    }
+
+    override fun matchesSafely(view: View): Boolean {
+      return view is ImageView && view.contentDescription.toString() == contentDescription
+    }
+  }
+
+  private fun withSeekBarPosition(position: Int) = object : TypeSafeMatcher<View>() {
+    override fun describeTo(description: Description) {
+      description.appendText("SeekBar with progress same as $position")
+    }
+
+    override fun matchesSafely(view: View): Boolean {
+      return view is SeekBar && view.progress == position
+    }
+  }
+
+  private fun clickSeekBar(position: Int): ViewAction {
+    return GeneralClickAction(Tap.SINGLE, object: CoordinatesProvider {
+      override fun calculateCoordinates(view: View?): FloatArray {
+        val seekBar = view as SeekBar
+        val screenPos = IntArray(2)
+        seekBar.getLocationInWindow(screenPos)
+        val trueWith = seekBar.width - seekBar.paddingLeft - seekBar.paddingRight
+
+        val relativePos = (position / seekBar.max).toFloat()
```

Comment:
Should this toFloat() happen before the division? Otherwise you're losing your fractional component.

Reply:
Also regarding the name, this is actually a ratio or a percentage not a position. A clearer name is a bit harder to come up with, though.